test/units/utils/test_unsafe_proxy.py

```
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# (c) 2018 Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.six import PY3
+from ansible.utils.unsafe_proxy import AnsibleUnsafe, AnsibleUnsafeText, UnsafeProxy, wrap_var
+
+import pytest
+
+
+def test_UnsafeProxy():
+    assert isinstance(UnsafeProxy({}), dict)
+    assert not isinstance(UnsafeProxy({}), AnsibleUnsafe)
+
+    assert isinstance(UnsafeProxy('foo'), AnsibleUnsafe)
+
+
+def test_wrap_var_string():
+    assert isinstance(wrap_var('foo'), AnsibleUnsafe)
+    assert isinstance(wrap_var(u'foo'), AnsibleUnsafe)
+    if PY3:
+        assert not isinstance(wrap_var(b'foo'), AnsibleUnsafe)
```

```suggestion
        assert isinstance(wrap_var(b'foo'), type(b''))
```

---

I used the `not isinstance` syntax, so that if someone were to implement something like:

```
class AnsibleUnsafeBytes(binary_type, AnsibleUnsafe):
    pass
```

The tests would have to be knowingly updated, because they would fail.

The change you recommend, would silently pass, due to making an assumption about the data.

The current method would fail, as it causes a change in behavior.

---

Ah, that makes sense. It would still be good to add the assertions while keeping the existing ones. In line with that, should most of the `isinstance` assertions which aren't negated check for `AnsibleUnsafeText` instead of just `AnsibleUnsafe`?

---

Feedback should be addressed.